### DISCUSSION TITLE
Interpolating at SX or MX variable?

### DISCUSSION THREAD

**Discussion by hurak** [casadi/casadi#3536] [Q&A] [answered]
It is emphasised in the [documentation for 1D lookup tables](https://web.casadi.org/docs/#d-lookup-tables) that "the `grid` and `values` arguments to `interpolant` must be numerical in nature". Let me invoke here the (Matlab) code from the documentation for convenience before asking the question:
```matlab
xgrid = 1:6;
V = [-1 -1 -2 -3 0 2];
lut = casadi.interpolant('LUT','bspline',{xgrid},V);
lut(2.5)
```

So, both `xgrid` and `V` must be numerical. OK. But can the `lut` be then evaluated at SX or MX variables? Obviously I am building a problem in which one of the optimization optimization variables affects the rest in a nonlinear way for which I only have a lookup table.

**Comment by LouisThomasRoy** [comment, marked as answer]
Hi @hurak, I have found that the interpolation only works with MX variables.

**Reply by jrbull** [reply]
Can you give an example? I have a model based on SX in which I want to interpolate a symbolic expression.

**Reply by josipkh** [reply]
Interpolation seems to work with SX now as well, I assume from v3.7.0.
I don't have a minimal example to share, but in my case switching from MX to SX caused no difference (used in an MPC framework).
